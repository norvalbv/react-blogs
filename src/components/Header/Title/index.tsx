import React, { ReactElement } from 'react';
import { DefTheme } from 'types';
import { getClassName, isLabelInProps } from 'utils';

export type TitleProps = {
  children?: string | ReactElement;
  className?: string;
  /**
   * Default 2.
   */
  level?: 1 | 2 | 3;
  theme?: DefTheme;
  props?: unknown;
};

const Title = ({
  children,
  className,
  level = 2,
  theme,
  ...props
}: TitleProps): ReactElement | null => {
  if (!children) {
    return null;
  }

  if (level === 1) {
    return (
      <h1 className={getClassName({ tag: 'h1', theme, className })} {...isLabelInProps(props)}>
        {children}
      </h1>
    );
  }

  if (level === 2) {
    return (
      <h2 className={getClassName({ tag: 'h2', theme, className })} {...isLabelInProps(props)}>
        {children}
      </h2>
    );
  }

  return (
    <h3 className={getClassName({ tag: 'h3', theme, className })} {...isLabelInProps(props)}>
      {children}
    </h3>
  );
};

export default Title;
