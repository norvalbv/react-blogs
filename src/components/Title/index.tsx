import { ReactElement } from 'react';
import { getClassName, isLabelInProps } from 'utils';

export type TitleProps = {
  children?: string | ReactElement;
  className?: string;
  /**
   * Default 2.
   */
  level?: 1 | 2 | 3;
  props?: unknown;
};

const Title = ({ children, className, level = 2, ...props }: TitleProps): ReactElement | null => {
  if (!children) {
    return null;
  }

  if (level === 1) {
    return (
      <h1 className={getClassName({ tag: 'h1', className })} {...isLabelInProps(props)}>
        {children}
      </h1>
    );
  }

  if (level === 2) {
    return (
      <h2 className={getClassName({ tag: 'h2', className })} {...isLabelInProps(props)}>
        {children}
      </h2>
    );
  }

  return (
    <h3 className={getClassName({ tag: 'h3', className })} {...isLabelInProps(props)}>
      {children}
    </h3>
  );
};

export default Title;
