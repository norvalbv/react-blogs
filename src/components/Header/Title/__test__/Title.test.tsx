import React, { FC } from 'react';
import renderer from 'react-test-renderer';

import Title, { TitleProps } from '..';

const TitleWithAllProps: FC<Required<TitleProps>> = Title;

describe('<Title />', () => {
  test('should render nothing if title is not provided', () => {
    const tree = renderer.create(<Title />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render correctly with all props', () => {
    const tree = renderer
      .create(
        <TitleWithAllProps testId="Test ID" className="text-h2" level={1}>
          Test title
        </TitleWithAllProps>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
