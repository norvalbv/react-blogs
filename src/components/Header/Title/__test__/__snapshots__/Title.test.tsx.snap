// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render correctly with all props 1`] = `
<h1
  className="text-h2"
  data-testid="Test ID"
>
  Test title
</h1>
`;

exports[`<Title /> should render nothing if title is not provided 1`] = `null`;
